jupyter-on-gke/jupyterhub.tf

```diff
@@ -26,111 +36,100 @@ provider "helm" {
-resource "google_iap_brand" "project_brand" {
-  count             = var.add_auth && var.add_auth && var.brand == "" ? 1 : 0
-  support_email     = var.support_email
-  application_title = "Cloud IAP protected Application"
-  project           = var.project_id
-}
-
-# Creates the OAuth client used in IAP
+# IAP Section: Creates the OAuth client used in IAP
 resource "google_iap_client" "iap_oauth_client" {
   count        = var.add_auth && var.client_id == "" ? 1 : 0
   display_name = "Jupyter-Client"
-  brand        = "projects/${data.google_project.project.number}/brands/${data.google_project.project.number}"
-}
-
-resource "kubernetes_namespace" "namespace" {
-  count = var.create_namespace ? 1 : 0
-  metadata {
-    labels = {
-      namespace = var.namespace
-    }
-
-    name = var.namespace
-  }
+  brand        = var.brand == "" ? "projects/${data.google_project.project.number}/brands/${data.google_project.project.number}" : var.brand
 }
 
+# IAP Section: Creates the GKE components
 module "iap_auth" {
   count  = var.add_auth ? 1 : 0
   source = "./iap_module"
 
-  project_id      = var.project_id
-  namespace       = var.namespace
-  service_name    = var.service_name
-  client_id       = var.client_id != "" ? var.client_id : google_iap_client.iap_oauth_client[0].client_id
-  client_secret   = var.client_id != "" ? var.client_secret : google_iap_client.iap_oauth_client[0].secret
-  url_domain_addr = var.url_domain_addr
-  url_domain_name = var.url_domain_name
-
-  depends_on = [
-    helm_release.jupyterhub,
-    kubernetes_namespace.namespace,
-    module.workload_identity_service_account
-  ]
+  project_id              = var.project_id
+  namespace               = var.namespace
+  default_backend_service = var.default_backend_service
+  service_name            = var.service_name
+  client_id               = var.client_id != "" ? var.client_id : google_iap_client.iap_oauth_client[0].client_id
+  client_secret           = var.client_id != "" ? var.client_secret : google_iap_client.iap_oauth_client[0].secret
+  url_domain_addr         = var.url_domain_addr
+  url_domain_name         = var.url_domain_name
+  depends_on              = [google_project_service.project_service]
 }
 
 module "workload_identity_service_account" {
   source = "./service_accounts_module"
 
   project_id      = var.project_id
   namespace       = var.namespace
-  service_account = "jupyter-service-account"
+  service_account = "jupyter-service-account1"
 }
 
 resource "helm_release" "jupyterhub" {
-  name            = "jupyterhub"
-  repository      = "https://jupyterhub.github.io/helm-chart"
-  chart           = "jupyterhub"
-  namespace       = var.namespace
-  cleanup_on_fail = "true"
+  name             = "jupyterhub"
+  repository       = "https://jupyterhub.github.io/helm-chart"
+  chart            = "jupyterhub"
+  namespace        = var.namespace
+  create_namespace = !contains(data.kubernetes_all_namespaces.allns.namespaces, var.namespace)
```

Comment:
Where is this var being used?

Reply:
used by helm, if namespace creation is required or not.